wordpress-mobile/WordPress-Android
# Jetpack Focus: Add Jetpack feature card show rule

Parent #17336 

This PR implements the show "rules" for the JetpackFeatureCard.
Includes:
- Replace todos with comments. 
- Removes the detekt ForbiddenComment annotation from MySiteViewModel
- Refactor: Moves the show logic from `MySiteViewModel` to `JetpackFeatureCardHelper` (and associated tests)

**Not Included**
CTA's for the card (showing the overlay or removing the cards from the menu after menu item actions). They will be handled in a separate PR.

**Merge Instructions**
1 - Ensure that #17681 has been merged
2 - Remove "Not ready for merge" label
3 - Merge as normal

**To test:**
Test 1: Card is not shown within Jetpack App
- Install JP app
- Login 
- Navigate to Me -> App Settings -> Debug Settings
- If `jp_removal_three` is enabled, please disable and restart the app
- Navigate to My Site
- ✅ Verify that the Jetpack Feature card is not shown

Test 2: Card is not shown with WordPress App when phase 3 flag is off
- Install WP app
- Login 
- Navigate to Me -> App Settings -> Debug Settings
- If `jp_removal_three` is enabled, please disable and restart the app
- Navigate to My Site
- ✅ Verify that the Jetpack Feature card is shown

Test 3: Card is shown when WordPress App and phase 3 flag is on
- Install WP app
- Login 
- Navigate to Me -> App Settings -> Privacy Settings and turn on Collection Information
- Navigate to -> Debug Settings -> 
- Enable `jp_removal_three` and restart the app
- Navigate to My Site
- ✅ Verify that the Jetpack Feature card is shown

## Regression Notes
1. Potential unintended areas of impact
The Jetpack feature card shows when it should not 

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and unit testing

3. What automated tests I added (or what prevented me from doing so)
Added `JetpackFeatureCardHelperTest`

PR submission checklist:
